### PR TITLE
Add Docs For SGLang Native Router

### PR DESCRIPTION
**Pull request by zhaochenyang20**
## Motivation

I added an official document for SGLang native router. Build from `https://github.com/sgl-project/sglang/tree/main/rust`.

## Checklist

- [x] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [x] Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [x] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

File: scripts/ci_install_dependency.sh [hunk 1]
```diff
@@ -19,3 +19,6 @@ pip install cutex
 
 # For compling xgrammar kernels
 pip install cuda-python nvidia-cuda-nvrtc-cu12
+
+# For router
+pip install sglang-router
```
Comment (ByronHsu) [inline scripts/ci_install_dependency.sh]: don't add this for now. the router is not well supported on all linux distribution yet. i have a separate ci test for router.

File: docs/index.rst [hunk 2]
```diff
@@ -39,6 +39,13 @@ The core features include:
    frontend/choices_methods.md
 
 
+.. toctree::
+   :maxdepth: 1
+   :caption: Native Router
```
Comment (ByronHsu) [inline docs/index.rst]: "native" router is a bit weird. Maybe "Router for Data Parallelism"?

File: scripts/ci_install_dependency.sh (hunk 1 above)
Comment (zhaochenyang20) [reply scripts/ci_install_dependency.sh]: Well. So how can we run the router in the docs? Can we make it pure markdown without running codes? @merrymercy

File: docs/index.rst (hunk 2 above)
Comment (zhaochenyang20) [reply docs/index.rst]: Native means that's SGLang official 😂 I stand for these words, and that means in the open source communities, we are all most unique.

Comment (merrymercy) [reply docs/index.rst]: I also feel we can drop "native" here

File: scripts/ci_install_dependency.sh (hunk 1 above)
Comment (merrymercy) [reply scripts/ci_install_dependency.sh]: yes, let us just do markdown instead of the interactive notebook.

**Review by merrymercy — CHANGES_REQUESTED** [review]
address comments?

**Comment by zhaochenyang20** [comment]
> address comments?

Yeah, I am working on this

**Review by ByronHsu — APPROVED** [review]
Thanks!

**Pull request merged**